src/sentry/discover/endpoints/discover_key_transactions.py

```diff
@@ -43,51 +45,143 @@ def get(self, request, organization):
+        if not self.has_team_feature(request, organization):
+            return Response(status=404)
+
+        transaction_name = request.GET.get("transaction")
+        if transaction_name is None:
+            raise ParseError(detail="A transaction name is required")
+
+        project = self.get_project(request, organization)
+
+        key_teams = TeamKeyTransaction.objects.filter(
+            organization=organization,
+            team__in=request.access.teams,
+            project=project,
+            transaction=transaction_name,
+        )
+
+        return Response(serialize(list(key_teams)), status=200)
+
     def post(self, request, organization):
         """ Create a Key Transaction """
         if not self.has_feature(request, organization):
             return Response(status=404)
 
         project = self.get_project(request, organization)
 
-        base_filter = {"organization": organization, "owner": request.user}
+        if not self.has_team_feature(request, organization):
+            base_filter = {"organization": organization, "owner": request.user}
+
+            with transaction.atomic():
+                serializer = serializers.KeyTransactionSerializer(
+                    data=request.data, context=base_filter
+                )
+                if serializer.is_valid():
+                    data = serializer.validated_data
+                    base_filter["transaction"] = data["transaction"]
+                    base_filter["project"] = project
+
+                    if KeyTransaction.objects.filter(**base_filter).exists():
+                        return Response(status=204)
+
+                    try:
+                        KeyTransaction.objects.create(**base_filter)
+                        return Response(status=201)
+                    # Even though we tried to avoid it, this KeyTransaction was created already
+                    except IntegrityError:
+                        return Response(status=204)
+                return Response(serializer.errors, status=400)
 
         with transaction.atomic():
-            serializer = KeyTransactionSerializer(data=request.data, context=base_filter)
+            serializer = serializers.TeamKeyTransactionSerializer(
+                data=request.data,
+                context={"mode": "create", "request": request},
+            )
+
             if serializer.is_valid():
                 data = serializer.validated_data
-                base_filter["transaction"] = data["transaction"]
-                base_filter["project"] = project
-
-                if KeyTransaction.objects.filter(**base_filter).exists():
+                base_filter = {
+                    "organization": organization,
+                    "project": project,
+                    "transaction": data["transaction"],
+                }
+
+                keyed_transactions = TeamKeyTransaction.objects.filter(
+                    **base_filter, team__in=data["team"]
+                )
```

Comment:
I think you can just do `.values_list("team_id", flat=True)` here?

Reply:
fixed in b9be6be